### PR TITLE
[Test] Add unit tests for srt/entrypoints/openai/

### PR DESCRIPTION
**Pull request by physics31415926**
## Summary

Add unit tests for `srt/entrypoints/openai/utils.py` and `usage_processor.py` as part of the unit test coverage improvement effort (#20865).

These modules contain pure logic for token usage calculation, logprobs conversion, embed tensor conversion, and cached token details processing — all untested prior to this PR.

### `test_usage_processor.py` (15 tests)
- `_details_if_cached`: boundary behavior (zero, negative, positive)
- `calculate_response_usage`: n_choices stride logic (only first choice per prompt counted for prompt_tokens)
- `calculate_streaming_usage`: `idx % n_choices == 0` filtering for prompt/cached tokens
- Edge cases: missing meta_info fields, empty dicts, cache report enable/disable

### `test_utils.py` (35 tests)
- `to_openai_style_logprobs`: tuple unpacking `(logprob, _, token_text)`, dict construction from top_logprobs, None entry handling
- `process_hidden_states_from_ret`: `len > 1` branch (returns last vs empty), None/missing/empty list
- `should_include_usage`: stream_options present/absent paths, OR logic with default
- `cached_tokens_details_from_dict`: `"storage" in details` branch routing
- `convert_embeds_to_tensors`: nesting depth detection (single vs batch), first_non_none search with leading None, dtype validation

## Test Results

```bash
# Command
PYTHONPATH=/root/sglang/python pytest test/registered/unit/entrypoints/openai/test_usage_processor.py test/registered/unit/entrypoints/openai/test_utils.py -v

# Output
======================== 50 passed, 1 warning in 9.45s =========================
```

## Checklist

- [x] Tests in `test/registered/unit/entrypoints/openai/` mirroring `srt/entrypoints/openai/`
- [x] Does NOT launch server or load model weights
- [x] Includes edge cases (None, empty list, negative count, leading None in batch, missing fields)
- [x] Uses `CustomTestCase` from `sglang.test.test_utils`
- [x] Registered with `register_cpu_ci()`
- [x] Locally tested and passing

Closes #20865

### CODE REVIEW

File: test/registered/unit/entrypoints/openai/test_utils.py [hunk 1]
```diff
@@ -0,0 +1,378 @@
...
+        self.assertEqual(result, [4, 5, 6])
+
+    def test_return_hidden_states_true_single(self):
+        """Test when return_hidden_states is True with single state."""
+        request = MagicMock(spec=ChatCompletionRequest)
+        request.return_hidden_states = True
+        ret_item = {"meta_info": {"hidden_states": [[1, 2, 3]]}}
+
+        result = process_hidden_states_from_ret(ret_item, request)
+        self.assertEqual(result, [])
```
[the remark this answers is not in the document]
Comment (physics31415926) [reply test/registered/unit/entrypoints/openai/test_utils.py:128]: This is intentional behavior. The hidden_states list contains one entry per token — the first entry is the prompt token's hidden state, and subsequent entries are generated tokens' hidden states. When len == 1, it means no generation token was produced yet, so returning [] (no generated hidden state to report) is correct. The same logic appears in the streaming path at serving_completions.py:367-370. This is not a bug.

File: test/registered/unit/entrypoints/openai/test_utils.py [hunk 2]
```diff
@@ -0,0 +1,378 @@
...
+        self.assertTrue(continuous)
+
+    def test_stream_options_overrides_default(self):
+        """Test that stream_options overrides default."""
+        stream_options = StreamOptions(include_usage=False, continuous_usage_stats=False)
+        include, continuous = should_include_usage(stream_options, True)
+        # Default is True, but stream_options.include_usage is False
+        # Result should be: include_usage OR default = False OR True = True
+        self.assertTrue(include)
+        self.assertFalse(continuous)
```
[the remark this answers is not in the document]
Comment (physics31415926) [reply test/registered/unit/entrypoints/openai/test_utils.py]: The OR logic here is by design. stream_response_default_include_usage is a server-level policy flag (set by the admin via --stream_response_default_include_usage), not a user-overridable default. Its semantics is "always include usage in streaming responses at minimum." Users can opt-in (include_usage=True) but cannot opt-out of a server-enforced policy. This matches the intended use case where operators want usage tracking regardless of client requests.
